Refactor/network node operation

0xxlegolas opened the pull request:
- merging burn and bringOnline in single function for NetworkNode
- removed bunch of params from createAndAnchorNetworkNode 
- added more params in depositFuel
- changed fuelTypeId  as smartObjectId to be consistent with other ids. 
- fixes naming convention from structures to assemblies

0xxlegolas committed:
Refactor Fuel System to Include Fuel Type in Deposit Functionality

**Changes:**
- Removed the `setFuelAmount` function as it was deemed unnecessary.
- Adjusted related tests to reflect the new deposit function signature and validate fuel type handling.
- Bring structure offline only if its online

This refactor enhances the fuel management system by allowing different fuel types to be deposited, improving the overall functionality and flexibility of the fuel system.

0xxlegolas committed:
refactor: rename NetworkStructureConnection to NetworkNodeAssemblyLink

- Rename table and update schema to use assemblyId
- Update related code and tests
- Align with game's assembly terminology

0xxlegolas committed:
refactor: enhance Deployable and Fuel systems for improved functionality

- Updated FuelSystem to include additional checks for fuel burn rate and efficiency, ensuring more accurate fuel consumption calculations.
- Refined energy requirement checks in NetworkNodeSystem to differentiate between connected and non-connected structures, improving energy management.

These changes aim to streamline the interaction between deployable structures and network nodes, enhancing overall system efficiency.

0xxlegolas committed:
fix: update DeployableSystem to handle smartObjectId checks

- This change ensures that the system correctly identifies and manages structures based on their connection status to network nodes

vayan commented on mud-contracts/world-v2/src/codegen/world/INetworkNodeSystem.sol:
```diff
@@ -27,11 +27,11 @@ interface INetworkNodeSystem {
-  function evefrontier__connectStructure(uint256 networkNodeId, uint256 structureId) external;
+  function evefrontier__connectStructure(uint256 networkNodeId, uint256 assemblyId) external;
 
-  function evefrontier__onStructureOnline(uint256 networkNodeId, uint256 structureId) external;
+  function evefrontier__onStructureOnline(uint256 networkNodeId, uint256 assemblyId) external;
 
-  function evefrontier__onStructureOffline(uint256 networkNodeId, uint256 structureId) external;
+  function evefrontier__onStructureOffline(uint256 networkNodeId, uint256 assemblyId) external;
```
do we also want to rename those? 

`evefrontier__connectAssembly`
`evefrontier__onAssemblyOnline`
`evefrontier__onAssemblyOffline`

0xxlegolas replied:
yes on it changing one by one haha

0xxlegolas committed:
feat: enhance Network Node and Deployable systems with connected assemblies management

- Added support for tracking connected assemblies in the Network Node, allowing for better management of structures linked to network nodes.
- Introduced new functions to handle the addition, retrieval, and updating of connected assemblies.
- Updated DeployableSystem to manage the offline state of connected structures when a network node goes offline.
- Refactored related functions in NetworkNodeSystem to ensure proper energy management and assembly disconnection.

These changes improve the overall functionality and efficiency of the network node system, enhancing the interaction between deployable structures and their energy sources.

0xxlegolas committed:
refactor: clean redundant data

- Removed unused fields from the schema in mud.config.ts, including reservedEnergy and operationStatus, to streamline the configuration.

0xxlegolas committed:
feat: add NetworkNodeEnergyHistory for tracking energy usage

- Introduced a new schema for NetworkNodeEnergyHistory in mud.config.ts to log historical energy usage by network nodes.

0xxlegolas committed:
build end to end tests

0xxlegolas committed:
merge burn and bringOnline for network node functionality

0xxlegolas committed:
Merge branch 'develop' into refactor/network-node-operation

0xxlegolas committed:
feat: enhance fuel management system with smart object integration

- Refactored FuelSystem to replace fuelTypeId with fuelSmartObjectId for consistency
- Adjusted Fuel and FuelEfficiencyConfig schemas to reflect the new smart object structure.
- Updated tests to validate the new fuel management logic and ensure proper functionality with the smart object IDs.

0xxlegolas committed:
Merge branch 'refactor/network-node-operation' of github.com:projectawakening/world-chain-contracts into refactor/network-node-operation

0xxlegolas committed:
refactor: rename structure references to assembly in NetworkNode and Deployable systems

0xxlegolas committed:
update abi artifacts

0xxlegolas committed:
feat: add WorldVersion table for Docker image version tracking

0xxlegolas committed:
refactor: update fuel management system and remove obsolete components
